heft-plugins/heft-typescript-plugin/src/TypeScriptBuilder.ts

```diff
@@ -1301,8 +1301,8 @@ function getFilesToTranspileFromBuilderProgram(
   builderProgram: TTypescript.BuilderProgram
 ): Map<string, string> {
   const changedFilesSet: Set<string> = (
-    builderProgram as unknown as { getState(): { changedFilesSet: Set<string> } }
-  ).getState().changedFilesSet;
+    builderProgram as unknown as { state: { changedFilesSet: Set<string> } }
+  ).state.changedFilesSet;
```

Comment:
To not break TS <5.6, you could just declare both and then use `state` if it's present, otherwise call `getState`.

Reply:
As in:

```diff ts
diff --git a/heft-plugins/heft-typescript-plugin/src/TypeScriptBuilder.ts b/heft-plugins/heft-typescript-plugin/src/TypeScriptBuilder.ts
index f1f4db242a..88e940d2f3 100644
--- a/heft-plugins/heft-typescript-plugin/src/TypeScriptBuilder.ts
+++ b/heft-plugins/heft-typescript-plugin/src/TypeScriptBuilder.ts
@@ -1300,9 +1300,13 @@ export class TypeScriptBuilder {
 function getFilesToTranspileFromBuilderProgram(
   builderProgram: TTypescript.BuilderProgram
 ): Map<string, string> {
-  const changedFilesSet: Set<string> = (
-    builderProgram as unknown as { getState(): { changedFilesSet: Set<string> } }
-  ).getState().changedFilesSet;
+  const builderProgramWithState = builderProgram as unknown as {
+    state?: { changedFilesSet: Set<string> };
+    getState?(): { changedFilesSet: Set<string> };
+  };
+
+  const changedFilesSet: Set<string> =
+    builderProgramWithState.state?.changedFilesSet ?? builderProgramWithState.getState!().changedFilesSet;
   const filesToTranspile: Map<string, string> = new Map();
   for (const fileName of changedFilesSet) {
     const sourceFile: TTypescript.SourceFile | undefined = builderProgram.getSourceFile(fileName);
```

Reply:
Unfortunately the repo can't update to 5.6 because this fix isn't released and the repo needs itself working on npm to publish :(

Reply:
Too right mate. Changes incoming